ding_periodical.module

```diff
@@ -135,12 +136,17 @@ function theme_ding_periodical_issues($variables) {
 
       // Check if the volume is reservable and add reservation button.
       if (!empty($issue_id) && ding_periodical_is_reservable($issues[$key])) {
-        $path = array($issue_id, $volume, $key);
+        $item_id = array($issue_id, $volume, $key);
+
+        foreach ($item_id as &$v) {
+          $v = rawurlencode($v);
+        }
+
         $button = array(
           array(
             '#theme' => 'link',
             '#text' => t('Reserve'),
-            '#path' => 'ting/object/' . $variables['entity']->id . '/reserve/' . implode('/', $path),
+            '#path' => 'ting/object/' . $variables['entity']->id . '/reserve/' . base64_encode(implode('|', $item_id)),
```

Comment:
Calling three times base64_encode in place and three time to decode in another - look weird, reservation link becomes three times longer.
Sort of:
implode('|', array(base64_encode(), base64_encode(), base64_encode()))